src/api/app/models/workflow/step/link_package_step.rb

```diff
@@ -5,25 +5,52 @@ class Workflow::Step::LinkPackageStep < ::Workflow::Step
   def call(options = {})
     return unless valid?
 
-    # FIXME: Support closed/merged/reopened PRs
-    return if scm_webhook.closed_merged_pull_request? || scm_webhook.reopened_pull_request?
-
     workflow_filters = options.fetch(:workflow_filters, {})
 
-    # Updated PR
-    if scm_webhook.updated_pull_request? && target_package.present?
-      update_subscriptions(target_package, workflow_filters)
-    else # New PR
+    case
+    when scm_webhook.new_pull_request?
+      create_link_package(workflow_filters)
+    when scm_webhook.updated_pull_request?
+      update_link_package(workflow_filters)
+    when scm_webhook.closed_merged_pull_request?
+      destroy_target_project
+    when scm_webhook.reopened_pull_request?
+      restore_target_project(workflow_filters)
+    end
+  end
+
+  private
+
+  def update_link_package(workflow_filters)
+    if target_package.blank?
```

Comment:
maybe we should have a method named `create_or_update_subscriptions` where we create subscriptions if they don't exist otherwise we update them.. what do you think @dmarcoux

Reply:
This method already exists in `Workflow::Step`. We also have `create_subscriptions` and `update_subscriptions`. I'm not sure if we either go with the separate methods or the single one which combines both. Do you have a preference?

https://github.com/openSUSE/open-build-service/blob/2acf0549256926c6b96190bab7a9f054e44ad447/src/api/app/models/workflow/step.rb#L87-L99